frontend/spec/features/checkout_confirm_insufficient_stock_spec.rb

```diff
@@ -10,11 +10,16 @@
 
     let(:order) { Spree::TestingSupport::OrderWalkthrough.up_to(:payment) }
     let(:order_product) { order.products.first }
-    let(:order_stock_item) { order_product.stock_items.first }
+
+    # This can't be memoized now in a `let` block, otherwise we
```

Comment:
So `order_stock_item`, which is memoized in the spec context, is updated to have some count on hand in the spec setup. However, the `order`s line item's stock items that we use in the spec (we stub `current_order` with it) are not updated in-memory. Since we don't access the database with the new code, we need to make `stock_item` a method so it's not memoized and everytime we access it we get the real deal.

Reply:
Ok, so alternatively we could refresh the in-memory order associations with `order.reload` after we set their count on hand in the specs, isn't it? Not a big fan of reloading objects, but I think that in the other specs we always preferred to use reload instead of creating methods to have fresh information. What do you think?

Reply:
Hm, this would help if we reloaded the object in the production code, but we don't actually do that - it's the other way around. I fiddled quite a bit to get this spec working; and I don't want to introduce a  `reload` in the production code...

Reply:
I meant to add a `order.reload` in the specs (on line 23 for example). I don't think we do need to do this in the runtime code because the update of the stock is not done on the same order's instance that is returned in the controller, or am I wrong?

Reply:
Yes, it is the same instance, that's the problem: https://github.com/solidusio/solidus/pull/4291/files#diff-4b1d3e2a52725ce1d5c20bb05a38a1184eec474deacfa6c58ef646e1a1b3b8aeR23

Reply:
It is the same instance in the tests, not in the production code, isn't it?

Reply:
For what is worth, I tried locally with a reload in the specs and it works without the new method or changing anything else.